detections/endpoint/multiple_users_failing_to_authenticate_from_one_process.yml

```diff
@@ -0,0 +1,49 @@
+name: Multiple Users Failing To Authenticate From One Process
+id: 9015385a-9c84-11eb-bef2-acde48001122
+version: 1
+date: '2021-04-13'
+author: Mauricio Velazco, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: The following analytic identifies a source process name failing to authenticate with multiple users. This behavior could represent an adversary
+ performing a Password Spraying attack against an Active Directory environment to obtain initial access or elevate privileges. 
+ In this logic, the number of unique users is hardcoded to '10' but can/should be customized to fit especific environments.
+ Event 4625 documents each and every failed attempt to logon to the local computer. This event generates on domain controllers, member servers, and workstations. Logon Type 2 
+ describes an iteractive logon attempt.\
+
+ This detection will trigger on the potenfially malicious host, perhaps controlled via a trojan or operated by an insider threat, from where a
+ password spraying attack is being executed. This could be a domain controller as well as a member server orworkstation.
+
+ The analytic's returned fields allow analysts to investigate the event further by providing fields like source process name, source account and attempted user accounts.
+
+search: ' `wineventlog_security` EventCode=4625 Logon_Type=2  !="-" | eval Source_Account = mvindex(Security_ID, 0) | eval Destination_Account = mvindex(Account_Name, 1)  | stats  dc(Destination_Account) AS unique_accounts values(Account_Name) as tried_accounts by Caller_Process_Name, Source_Account, ComputerName  | where  unique_accounts > 10 | `multiple_users_failing_to_authenticate_from_one_process_filter` '
```

Comment:
A couple of notes/suggestions - I'd go with simpler ones first, and then at the end suggest something about this detection as a whole:

* What does '!="-"' mean? Pease note that most events would have "-" field values. Please consider excluding this constraint. for details have a look at example events here: 
https://confluence.splunk.com/pages/viewpage.action?spaceKey=PROD&title=Active+Directory%3A+Identities+and+Locations+of+Logging+for+the+Entities+Reported+in+AD+events

* Why are me making a source user to be a Security_ID? Source user instead is Subject -> Account Name (in TA terminology this would be just Account Name). Now, for most part we won;t care about the source user, and moreover it won't be really reported in 4625s. Instead this field is usually populated with the name of the device where the auth failed (thus, we'd have a dest device here). Further, even for dest device, we have better fields in the event to make extraction - namely ComputerName field. Please see example events in the same confluence page.  

* Next, this part "values(Account_Name) as tried_accounts" makes little sense- let's just please count Destination_Accounts. Those are the accounts which are attempted and failed.

* Now for the main detection: Not sure you'd really want to do just "Logon_Type = 2", reasons being:
1) This will only detect User32 logons, i.e., the logons that access Windows's desktop/logon screen. However, this will exclude all other remote authentications (type 3 and type 8), such as picking up emails, accessing shared resources, remote execution, etc.
2) For Logon Type 2, caller process will always be the same:  C:\Windows\System32\winlogon.exe, so you won't be finding more processess. IN THE END, this may end up just raising noise as many users fail to login to a Windows desktop screen or a server - then this won't be very relevant or even precise. 

* Overall, this detection is not sufficient/insightful if we are doing just "User32 / winlogon.exe" authentication. On the other hand it may be very noise prone if we extend it to logon type 3 and 8, because then we'd be colliding with IT issues - failed servers, expired passwords ... stuff that happens a bunch with Outlook servers.
 
* Let's please meet and brainstorm how to make something better with this detection.

Reply:
- The first comment was a mistake. I just updated it, its: Caller_Process_Name!="-" . On my testing, I noticed that in some scenarios ( For local SYSTEM logon for example), there is no Caller_Process_Name.